### ISSUE TITLE
evaluating indicator 

### ISSUE THREAD

**Issue by QINGHAOcui** [LiuZhu-CV/CRMEF#2]
How is the evaluation index function defined during data validation, such as FSIM index. thank you very much indeed

**Comment by LiuZhu-CV** [comment]
import cv2
import os
import numpy as np
from skimage.metrics import peak_signal_noise_ratio as psnr
from skimage.metrics import structural_similarity as ski_ssim
import lpips
import torch
from torch.autograd import Variable

loss_fn_alex = lpips.LPIPS(net='alex') # best forward scores

methods = ['2024cl','2024wocl','GAL','HoLoCo','tpami_new','DeFusion','result_BHF',]
for method in methods:
    psnr_list = []
    ssim_list = []
    lpips_list = []
    files_1 = os.listdir('./'+method+'/')
    for file in files_1:
        img = cv2.imread('./'+method+'/'+file)
        file_ = file.split('low')[1]
        file2 = 'gt'+file_.split('.')[0]+'.png'
        img2 =cv2.imread('./trainCC/'+file)
        psnr_ = psnr(image_test=img, image_true=img2,data_range=255)
        ssim_ = ski_ssim(img,img2,channel_axis=2)
        image_o_y = img / 127.5 - 1
        image_gt_y = img2 / 127.5 - 1
        # tensor
        image_o_y2 = np.array(image_o_y)[np.newaxis, :]
        image_o_y2 = np.transpose(image_o_y2, (0, 3, 1, 2))
        # print(np.shape(im_input), im_input)
        image_o_y2 = torch.tensor(image_o_y2).type(torch.FloatTensor)
        image_o_y2 = Variable(image_o_y2, requires_grad=False)

        image_gt_y2 = np.array(image_gt_y)[np.newaxis, :]
        image_gt_y2 = np.transpose(image_gt_y2, (0, 3, 1, 2))
        # print(np.shape(im_input), im_input)
        image_gt_y2= torch.tensor(image_gt_y2).type(torch.FloatTensor)
        image_gt_y2 = Variable(image_gt_y2, requires_grad=False)

        d = loss_fn_alex(image_gt_y2, image_o_y2).detach().numpy()

        psnr_list.append(psnr_)
        ssim_list.append(ssim_)
        lpips_list.append(d)
    print(method,np.mean(psnr_list),np.mean(ssim_list), np.mean(lpips_list))

**Comment by LiuZhu-CV** [comment]
FSIM can be found at "https://github.com/nnUyi/DerainZoo"
